sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
+
+    return os;
+  }
+
+  private App getApp(PackageInfo packageInfo) {
+    App app = new App();
+    app.setIdentifier(packageInfo.packageName);
+    app.setStartTime(
+        DateUtils
+            .getCurrentDateTime()); // is it App. start? this was not supposed to be a lazy class?
+    //    app.setHash(); whats that?
+    //    app.setBuildType(); whats that? might be possible with BuildConfig.BUILD_VARIANT but Apps
+    // side, also for flavor
+    app.setName(getApplicationName());
+    app.setVersion(packageInfo.versionName);
+    app.setBuild(getVersionCode(packageInfo));
+
+    // App name and starttime don't depend on packageInfo, should we move outside of the packageInfo
+    // NPE check like the old the SDK?
+
+    return app;
+  }
+
+  /**
+   * Get the device's current kernel version, as a string. Attempts to read /proc/version, and falls
+   * back to the 'os.version' System Property.
+   *
+   * @return the device's current kernel version, as a string
+   */
+  private String getKernelVersion() {
+    // its possible to try to execute 'uname' and parse it or also another unix commands or even
+    // looking for well known root installed apps
+    String errorMsg = "Exception while attempting to read kernel information";
+    String defaultVersion = System.getProperty("os.version");
+
+    BufferedReader br = null;
+    try {
+      File file = new File("/proc/version");
+      if (!file.canRead()) {
+        return defaultVersion;
+      }
+
+      br = new BufferedReader(new FileReader(file));
+      return br.readLine();
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, errorMsg, e);
+    } finally {
+      if (br != null) {
+        try {
+          br.close();
+        } catch (IOException ioe) {
+          log(SentryLevel.ERROR, errorMsg, ioe);
+        }
+      }
+    }
+
+    return defaultVersion;
+  }
+
+  /**
+   * Attempt to discover if this device is currently rooted. From:
+   * https://stackoverflow.com/questions/1101380/determine-if-running-on-a-rooted-device
+   *
+   * @return true if heuristics show the device is probably rooted, otherwise false
+   */
+  private Boolean isRooted() {
+    // we could get some inspiration from https://github.com/scottyab/rootbeer
+    if (Build.TAGS != null && Build.TAGS.contains("test-keys")) {
+      return true;
+    }
+
+    String[] probableRootPaths = {
+      "/data/local/bin/su",
+      "/data/local/su",
+      "/data/local/xbin/su",
+      "/sbin/su",
+      "/su/bin",
+      "/su/bin/su",
+      "/system/app/SuperSU",
+      "/system/app/SuperSU.apk",
+      "/system/app/Superuser",
+      "/system/app/Superuser.apk",
+      "/system/bin/failsafe/su",
+      "/system/bin/su",
+      "/system/sd/xbin/su",
+      "/system/xbin/daemonsu",
+      "/system/xbin/su"
+    };
+
+    for (String probableRootPath : probableRootPaths) {
+      try {
+        if (new File(probableRootPath).exists()) {
```

Comment:
How much value `isRooted` gives us? This looks really expensive. If we are keeping it we should cache the result on first use.

Reply:
`isRooted` is only useful when you rely on features for rooted devices, not for normal applications usually, and yes, a bit expensive, definitely those things should be cached

Reply:
same as https://github.com/getsentry/sentry-android/pull/35#discussion_r336448620